### ISSUE TITLE
Python plug-ins not handling queries

### ISSUE THREAD

**Issue by AdamPS** [albertlauncher/plugins#84]
It appears that the `handleQuery()` function is not called. It's the same for every python plug-in I tested: my own and the built-in ones. C++ plug-ins are working.

For example in the API test plugin, I examine the log. The "initialize" log is shown. No logs are shown from handleQuery. If I type "delay" then it is not detected (it just does a web search).

Albert 0.18.11
Python 3.10.6
Ubuntu 22.04.1 LTS

**Comment by ManuelSchneid3r** [comment]
I know there's not issue template here. But please post Albert report and stdout. See Albert - h

**Comment by AdamPS** [comment]
Thanks.

```
$ albert -r
        Albert version: 0.18.11
            Build date: Jan 11 2023 09:39:58
            Qt version: 6.2.4
  QT_QPA_PLATFORMTHEME: 
       Binary location: /usr/bin/albert
                   PWD: /home/adam/main
                 SHELL: /bin/bash
                  LANG: en_GB.UTF-8
      XDG_SESSION_TYPE: x11
   XDG_CURRENT_DESKTOP: ubuntu:GNOME
       DESKTOP_SESSION: ubuntu-xorg
   XDG_SESSION_DESKTOP: ubuntu-xorg
            ICON THEME: Adwaita
                    OS: Ubuntu 22.04.1 LTS
     OS (type/version): ubuntu/22.04
             Build ABI: x86_64-little_endian-lp64
  Arch (build/current): x86_64/x86_64
 Kernel (type/version): linux/5.15.0-56-generic

```

**Comment by AdamPS** [comment]
```
albert -d
11:06:55 [debg:albert] Creating crash indicator file
11:06:55 [debg:albert] Checking for a running instance…
11:06:55 [debg:albert] Creating local socket /home/adam/.cache/albert_socket
11:06:55 [debg:albert] Initializing database…
11:06:55 [debg:albert] UsageHistory::mruScores    115 µs
11:06:55 [debg:albert] Searching native plugins in /usr/lib/x86_64-linux-gnu/albert
11:06:55 [debg:albert] Found valid native plugin /usr/lib/x86_64-linux-gnu/albert/libsnippets.so
11:06:55 [debg:albert] Found valid native plugin /usr/lib/x86_64-linux-gnu/albert/libterminal.so
11:06:55 [debg:albert] Found valid native plugin /usr/lib/x86_64-linux-gnu/albert/libsystem.so
11:06:55 [debg:albert] Found valid native plugin /usr/lib/x86_64-linux-gnu/albert/libcalculator.so
11:06:55 [debg:albert] Found valid native plugin /usr/lib/x86_64-linux-gnu/albert/liburlhandler.so
11:06:55 [debg:albert] Found valid native plugin /usr/lib/x86_64-linux-gnu/albert/libfiles.so
11:06:55 [debg:albert] Found valid native plugin /usr/lib/x86_64-linux-gnu/albert/libpython.so
11:06:55 [debg:albert] Found valid native plugin /usr/lib/x86_64-linux-gnu/albert/libssh.so
11:06:55 [debg:albert] Found valid native plugin /usr/lib/x86_64-linux-gnu/albert/libapplications.so
11:06:55 [debg:albert] Found valid native plugin /usr/lib/x86_64-linux-gnu/albert/libchromium.so
11:06:55 [debg:albert] Found valid native plugin /usr/lib/x86_64-linux-gnu/albert/libwidgetsboxmodel.so
11:06:55 [debg:albert] Found valid native plugin /usr/lib/x86_64-linux-gnu/albert/libwebsearch.so
11:06:55 [debg:albert] Found valid native plugin /usr/lib/x86_64-linux-gnu/albert/libhash.so
11:06:55 [info:albert] Hotkey set to Ctrl+Space
11:06:55 [debg:albert] Loading frontend plugin…
11:06:56 [info:albert] Loading plugin system
11:06:56 [debg:albert] [1ms] Successfully loaded 'system'
11:06:56 [info:albert] Loading plugin calculator
11:06:56 [debg:albert] [1ms] Successfully loaded 'calculator'
11:06:56 [info:albert] Loading plugin urlhandler
11:06:56 [debg:albert] [1ms] Successfully loaded 'urlhandler'
11:06:56 [info:albert] Loading plugin files
11:06:56 [info:files] Indexing XXXX
11:06:56 [debg:albert] [6ms] Successfully loaded 'files'
11:06:56 [info:albert] Loading plugin python
11:06:56 [info:python] Python version: 3.10.6 (main, Nov 14 2022, 16:10:14) [GCC 11.3.0]
11:06:56 [debg:python] Searching Python plugins in /home/adam/.local/share/albert/python/plugins
11:06:56 [debg:python] Python package init file does not exist /home/adam/.local/share/albert/python/plugins/__pycache__
11:06:56 [debg:python] Found valid Python plugin /home/adam/.local/share/albert/python/plugins/bluetooth.py
11:06:56 [debg:python] Found valid Python plugin /home/adam/.local/share/albert/python/plugins/bookmarks.py
11:06:56 [debg:python] Found valid Python plugin /home/adam/.local/share/albert/python/plugins/drupal.py
11:06:56 [debg:python] Searching Python plugins in /usr/share/albert/python/plugins
11:06:56 [debg:python] Found valid Python plugin /usr/share/albert/python/plugins/api_test
11:06:56 [debg:python] Found valid Python plugin /usr/share/albert/python/plugins/arch_wiki
11:06:56 [debg:python] Found valid Python plugin /usr/share/albert/python/plugins/aur
11:06:56 [debg:python] Found valid Python plugin /usr/share/albert/python/plugins/copyq
11:06:56 [debg:python] Found valid Python plugin /usr/share/albert/python/plugins/googletrans
11:06:56 [debg:python] Found valid Python plugin /usr/share/albert/python/plugins/pass
11:06:56 [debg:python] Found valid Python plugin /usr/share/albert/python/plugins/pomodoro
11:06:56 [debg:python] Found valid Python plugin /usr/share/albert/python/plugins/python_eval
11:06:56 [debg:python] Path is not a python file /usr/share/albert/python/plugins/README.md
11:06:56 [debg:python] Found valid Python plugin /usr/share/albert/python/plugins/trash
11:06:56 [debg:python] Found valid Python plugin /usr/share/albert/python/plugins/wikipedia
11:06:56 [info:albert] Loading plugin api_test
11:06:56 [info:python] initialize
11:06:56 [debg:albert] [1ms] Successfully loaded 'api_test'
11:06:56 [debg:albert] [55ms] Successfully loaded 'python'
11:06:56 [info:albert] Loading plugin ssh
11:06:56 [debg:albert] [0ms] Successfully loaded 'ssh'
11:06:56 [info:albert] Loading plugin applications
11:06:56 [debg:albert] [0ms] Successfully loaded 'applications'
11:06:56 [info:albert] Loading plugin websearch
11:06:56 [debg:albert] [0ms] Successfully loaded 'websearch'
11:06:56 [debg:applications] Indexing desktop file: XXXX
11:06:56 [info:files] Indexing /data/music
11:06:56 [debg:albert] TIME:     71 µs ['globalsearch':'']
11:06:56 [info:files] Indexing XXXX
11:07:34 [debg:albert] Received message: toggle
11:07:34 [debg:albert] Query deleted. [#0 '']
11:07:34 [debg:albert] TIME:     18 µs ['globalsearch':'']
11:07:34 [debg:albert] Query deleted. [#1 '']
11:07:34 [debg:albert] TIME:    106 µs ['albert':'a']
11:07:34 [debg:albert] TIME:     90 µs ['plugins':'a']
11:07:34 [debg:albert] TIME:     36 µs ['ssh':'a']
11:07:34 [debg:albert] TIME:     11 µs ['urlhandler':'a']
11:07:34 [debg:albert] TIME:    153 µs ['files':'a']
11:07:34 [debg:albert] TIME:     27 µs ['system':'a']
11:07:34 [debg:albert] TIME:     16 µs ['websearch':'a']
11:07:34 [debg:albert] TIME:     64 µs ['applications':'a']
11:07:34 [debg:albert] TIME:    867 µs ['calculator':'a']
11:07:34 [debg:albert] TIME:   1447 µs ['globalsearch':'a']
11:07:34 [debg:albert] Query deleted. [#2 'a']
11:07:34 [debg:albert] TIME:     68 µs ['albert':'aa']
11:07:34 [debg:albert] TIME:     87 µs ['plugins':'aa']
11:07:34 [debg:albert] TIME:     33 µs ['ssh':'aa']
11:07:34 [warn:default] QItemSelectionModel: Selecting when no model has been set will result in a no-op.
11:07:34 [debg:albert] TIME:     13 µs ['urlhandler':'aa']
11:07:34 [debg:albert] TIME:     86 µs ['applications':'aa']
11:07:34 [debg:albert] TIME:     65 µs ['calculator':'aa']
11:07:34 [debg:albert] TIME:      7 µs ['websearch':'aa']
11:07:34 [debg:albert] TIME:     52 µs ['files':'aa']
11:07:34 [debg:albert] TIME:     38 µs ['system':'aa']
11:07:34 [debg:albert] TIME:    419 µs ['globalsearch':'aa']

```

**Comment by AdamPS** [comment]
I typed 'aa'. The python plug-in api_test is not logged in the list of plug-ins handling the query. There is a warning but it doesn't seem very relevant.

Many thanks

**Comment by ManuelSchneid3r** [comment]
@AdamPS I am not a magician. You have to post the warnings.

**Comment by AdamPS** [comment]
Thanks

> You have to post the warnings.
I already posted the entire log above. There is one warning near to the end.

**Comment by ManuelSchneid3r** [comment]
Puthon plugins are no global query handlers. They have to be triggered. This differentiation has been introduced in 0.18. Python can not be multithreaded by cpython design so they should not be run by default. Global query handling is a C++ only feature atm. I spent days trying to get this into python. Even adjusted the C++ interface for it. i am not even convinced that it is a good idea. Closing since this is technically not a bug.
